plugin/src/main/java/com/denizenscript/denizen/events/block/SpongeAbsorbsScriptEvent.java

```diff
@@ -0,0 +1,62 @@
+package com.denizenscript.denizen.events.block;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import org.bukkit.block.BlockState;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.SpongeAbsorbEvent;
+
+public class SpongeAbsorbsScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // sponge absorbs
+    //
+    // @Location true
+    //
+    // @Group Block
+    //
+    // @Warning this event may in some cases double-fire, requiring usage of the 'ratelimit' command (like 'ratelimit <script> 1t') to prevent doubling actions.
+    //
+    // @Cancellable true
+    //
+    // @Triggers when Sponge absorbs water.
+    //
+    // @Context
+    // <context.location> returns the location of the Sponge.
+    // <context.blocks> returns a ListTag of blocks that are being removed.
+    //
+    // -->
+
+    public SpongeAbsorbsScriptEvent() {
+        registerCouldMatcher("sponge absorbs");
+    }
+
+    public SpongeAbsorbEvent event;
+    public LocationTag location;
+    public ListTag blocks;
+
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "location" -> location;
+            case "blocks" -> blocks;
```

Comment:
You don't need to set a `location` or `blocks` variable because it's not used anywhere else but here. You can access the location and the blocks from the `event` variable (using the `.getBlock().getLocation()` and `.getBlocks()` methods, respectively). You can just take the assignment code from the `onSpongeAbsorbEvent` function and put it here.